runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -1208,11 +1209,10 @@ int32_t TR_EscapeAnalysis::performAnalysisOnce()
             else
                {
                TR_OpaqueClassBlock *clazz = (TR_OpaqueClassBlock*)classNode->getSymbol()->castToStaticSymbol()->getStaticAddress();
-
-               if (TR::Compiler->cls.isPrimitiveValueTypeClass(clazz))
+               if (clazz && comp()->fej9()->getNullRestrictedArrayClassFromComponentClass(clazz))
                   {
                   if (trace())
-                     traceMsg(comp(), "   Fail [%p] because array has primitive value type elements\n", candidate->_node);
+                     traceMsg(comp(), "   Fail [%p] because array could be null-restricted array\n", candidate->_node);
                   rememoize(candidate);
                   _candidates.remove(candidate);
                   }
```

Comment:
I think this entire block can go away.  The array allocation will always be a nullable array.

Reply:
Addressed in [7bc96d5](https://github.com/eclipse-openj9/openj9/compare/9fd2f09e62d10c0d31fb534bace077e933045ec4..7bc96d56b54a0663b220ede073dc58ef195f2dbe)